### ISSUE TITLE
dsv.format could take an optional columns array?

### ISSUE THREAD

**Issue by mariocasciaro** [d3/d3-dsv#12]
Sometimes it is useful to explicitly specify a column order for the generated CSV file, other times we may want to exclude one or more columns, moreover, providing a column list in input will avoid the initial scan of the row list required to extract the column names.
I would mainly implement it for the first reason, since a csv file is often opened using a spreadsheet app like Excel, where the order of the columns should not be just random.

If you are interested, I can work on a PR, since I need the feature anyway.

Thanks for your time and for the lib!

**Comment by mbostock** [comment]
This is why there’s a [*dsv*.formatRows](https://github.com/d3/d3-dsv#dsv_formatRows) method. I’ve added some examples to the API reference. Hope this helps!

**Comment by mariocasciaro** [comment]
Thanks for the answer, it's all clear now

**Comment by mbostock** [comment]
Even though you *can* do this with *dsv*.formatRows, I think it would be easy to support this, so I’m reopening this issue. As you suggested, the [*dsv*.format](https://github.com/d3/d3-dsv#dsv_format) method could take an optional array of column names. Then, instead of this:

```js
var string = csv.formatRows([[
    "year",
    "make",
    "model",
    "length"
  ]].concat(data.map(function(d, i) {
  return [
    d.year,
    d.make,
    d.model,
    d.length
  ];
})));
```

You could say this:

```js
var string = csv.format(data, ["year", "make", "model", "length"]);
```

Which is obviously much nicer. :grin:

Another possibility is that we could check if the passed-in *rows* array exposes a `columns` array, now that *dsv*.parse supports this feature. But I think this is probably too magical, since you might forget about the existence of the columns property after transforming the array. So, better to make it an explicit argument.

**Comment by mbostock** [comment]
Fixed in 0.1.11.

**Comment by mariocasciaro** [comment]
Thanks for the feature! It's definitely a much neater approach
